cmd/maya-exporter/app/collector/pool/pool_test.go

```diff
@@ -0,0 +1,286 @@
+				regexp.MustCompile(`openebs_zpool_list_parse_error_count 4`),
+			},
+		},
+	}
+
+	for name, tt := range cases {
+		t.Run(name, func(t *testing.T) {
+			runner = tt.run
+			pool := New()
+			if err := prometheus.Register(pool); err != nil {
+				t.Fatalf("collector failed to register: %s", err)
+			}
+			defer prometheus.Unregister(pool)
+
+			server := httptest.NewServer(promhttp.Handler())
+			defer server.Close()
+
+			client := http.DefaultClient
+			client.Timeout = 5 * time.Second
+			resp, err := client.Get(server.URL)
+			if err != nil {
+				t.Fatalf("unexpected failed response from prometheus: %s", err)
+			}
+			defer resp.Body.Close()
+
+			buf, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				t.Fatalf("failed reading server response: %s", err)
+			}
+
+			for _, re := range tt.match {
+				if !re.Match(buf) {
+					fmt.Println(string(buf))
+					t.Errorf("failed matching: %q", re)
+				}
+			}
+
+			for _, re := range tt.unmatch {
+				if !re.Match(buf) {
+					t.Errorf("failed unmatching: %q", re)
+				}
+			}
+		})
+	}
+}
+
+func TestGetInitStatus(t *testing.T) {
+	cases := map[string]struct {
+		run   testRunner
+		count int
+	}{
+		"Test0": {
+			run: testRunner{
+				stdout: []byte(`
+				pool: mypool
+				state: ONLINE
+				 scan: none requested
+			   config:
+
+				   NAME                                  STATE     READ WRITE CKSUM
+				   mypool                                ONLINE       0     0     0
+					 raidz1-0                            ONLINE       0     0     0
+					   /home/infinity/experiments/pool1  ONLINE       0     0     0
+					   /home/infinity/experiments/pool2  ONLINE       0     0     0
+					 raidz1-1                            ONLINE       0     0     0
+					   /home/infinity/experiments/pool3  ONLINE       0     0     0
+					   /home/infinity/experiments/pool4  ONLINE       0     0     0`),
+			},
+		},
+		"Test1": {
+			run: testRunner{
+				isError: true,
+			},
+			count: 3,
+		},
+		"Test2": {
+			run: testRunner{
+				isError: true,
+			},
+			count: 1,
+		},
+	}
+	for name, tt := range cases {
+		t.Run(name, func(t *testing.T) {
+			runner = tt.run
+			pool := New()
+			count = tt.count
+			pool.GetInitStatus(1 * time.Second)
+		})
+	}
+}
+
+func TestRejectRequestCounter(t *testing.T) {
+	reqCount := 100
+	output := regexp.MustCompile(`openebs_zpool_reject_request_count\s\d+`)
+	runner = testRunner{
+		stdout: []byte("cstor-5ce4639a-2dc1-11e9-bbe3-42010a80017a	3000	23423 1341	-	0	iauwb	1.00 REMOVED	-"),
+	}
+	col := New()
+	if err := prometheus.Register(col); err != nil {
+		t.Fatalf("collector failed to register: %s", err)
+	}
+
+	server := httptest.NewServer(promhttp.Handler())
+	var body io.ReadCloser
+
+	wg := sync.WaitGroup{}
+	wg.Add(reqCount)
+	for i := 0; i < reqCount; i++ {
+		go func(server *httptest.Server) {
+			defer wg.Done()
+			client := http.DefaultClient
+			client.Timeout = 5 * time.Second
+			resp, err := client.Get(server.URL)
+			body = resp.Body
```

Comment:
here, variable `body' seems to be used across multiple go-routines. Is that fine?

Reply:
yes